src/Infrastructure/Services/Github/GithubGraphQLService.cs

```diff
@@ -0,0 +1,509 @@
+        var stats = new CommitStatistics();
+
+        _logger.LogInformation(
+            "Processing {TotalRepos} repositories in batches of {BatchSize} for user {Username}",
+            allRepos.Count,
+            DefaultBatchSize,
+            username);
+
+        for (var i = 0; i < allRepos.Count; i += DefaultBatchSize)
+        {
+            var batch = allRepos.Skip(i).Take(DefaultBatchSize).ToList();
+            var batchStats = await GetCommitStatisticsForBatch(client, username, batch, since, cancellationToken);
+
+            stats.CommitsCount += batchStats.CommitsCount;
+            stats.FilesAdjusted += batchStats.FilesAdjusted;
+            stats.ChangesInFilesCount += batchStats.ChangesInFilesCount;
+            stats.AdditionsInFilesCount += batchStats.AdditionsInFilesCount;
+            stats.DeletionsInFilesCount += batchStats.DeletionsInFilesCount;
+
+            _logger.LogDebug(
+                "Processed batch {BatchNumber}/{TotalBatches} for user {Username}",
+                (i / DefaultBatchSize) + 1,
+                (allRepos.Count + DefaultBatchSize - 1) / DefaultBatchSize,
+                username);
+        }
+
+        return stats;
+    }
+
+    private async Task<CommitStatistics> GetCommitStatisticsForBatch(
+        GraphQLHttpClient client,
+        string username,
+        List<Repository> repositories,
+        string since,
+        CancellationToken cancellationToken)
+    {
+        if (repositories.Count == 0)
+        {
+            return new CommitStatistics();
+        }
+
+        // Build a dynamic GraphQL query for the batch
+        var repoQueries = new List<string>();
+        var variables = new Dictionary<string, object>
+        {
+            ["username"] = username,
+            ["since"] = since
+        };
+
+        for (var i = 0; i < repositories.Count; i++)
+        {
+            var repo = repositories[i];
+            var alias = $"repo{i}";
+            variables[$"owner{i}"] = repo.Owner.Login;
+            variables[$"name{i}"] = repo.Name;
+
+            repoQueries.Add($@"
+                {alias}: repository(owner: $owner{i}, name: $name{i}) {{
+                  defaultBranchRef {{
+                    target {{
+                      ... on Commit {{
+                        history(first: 100, since: $since) {{
+                          totalCount
+                          nodes {{
+                            committedDate
+                            additions
+                            deletions
+                            changedFiles
+                          }}
+                        }}
+                      }}
+                    }}
+                  }}
+                }}");
+        }
+
+        var batchQuery = new GraphQLRequest
+        {
+            Query = $@"
+                query GetCommitsBatch($username: String!, $since: GitTimestamp!, {string.Join(", ", repositories.Select((r, i) => $"$owner{i}: String!, $name{i}: String!"))}) {{
```

Comment:
The following error appears: 'Variable $username is declared by GetCommitsBatch but not used'. Please fix

Reply:
new error appears:  The requested operation requires an element of type 'Object', but the target element has type 'Array'. please fix

Reply:
By the way, still empty data Is being returned. Seems like the query is incorrect. Please fix it

Reply:
There is a code https://github.com/AntonProkopyev/gh_summary_bot/blob/master/gh_summary_bot/github_source.py#L161 where a proper query is written. Analyze it and adjust my code to make it work

Reply:
Please rewrite code from python example to .net